pulpcore/cli/ansible/repository.py

```diff
@@ -8,20 +8,38 @@
     PulpAnsibleRoleRemoteContext,
 )
 from pulpcore.cli.common.context import (
-    EntityDefinition,
     PulpContext,
     PulpRepositoryContext,
     pass_pulp_context,
     pass_repository_context,
 )
 from pulpcore.cli.common.generic import (
-    destroy_by_name,
-    list_entities,
-    show_by_name,
+    create_command,
+    destroy_command,
+    href_option,
+    label_command,
+    label_select_option,
+    list_command,
+    name_option,
+    show_command,
+    update_command,
     version_command,
 )
 
 
+def _remote_callback(
+    ctx: click.Context, param: click.Parameter, value: Optional[str]
+) -> Optional[str]:
+    # Pass None and "" verbatim
+    if value:
+        pulp_ctx: PulpContext = ctx.find_object(PulpContext)
+        try:
+            return PulpAnsibleCollectionRemoteContext(pulp_ctx, entity={"name": value}).pulp_href
+        except click.ClickException:
+            return PulpAnsibleRoleRemoteContext(pulp_ctx, entity={"name": value}).pulp_href
```

Comment:
If it's lazy there is no exception created, it'll always return a CollectionRemoteContext.  I would need to refactor these contexts even more if I wanted it to be lazy.

Reply:
I mean you make it lazy by not reading the `pulp_href` property here. It will still return the context according to the specified type, but the actual lookup (api call) will be deferred to `preprocess_entity` which is only called from inside the `create`/`update` function.

Reply:
I tried it the way you suggested and it didn't work.  When a user specifies the remote name there is no way for the CLI to know if it is a role or collection remote without creating a context and searching for the entity. Upon that search is where the exception is raise and thus the type is figured out. Yes there is a bad scenario where if you mistype the remote name and it's a collection remote you where trying for the error message would return 'role remote not found', but without refactoring more things it's not easy to find out what type the remote is that the user wants.

Reply:
Ah, again on my pr (for the container stuff) we have a repository-type option. It is markde is_eager, so the information is available, once this option is evaluated.

Reply:
https://github.com/pulp/pulp-cli/pull/140/files#diff-b8e25b575de53e02e0f0ef7ced1bda086d47668caaf68bfd6450b42f2ca7d7b9R47
I'm not so keen to use this meta dict, but that is the solution i came up with, and the `--repository-type` option was already there.

Reply:
Having a common way to reference Pulp objects like discussed in the meeting would solve this issue.